src/zarr/core/metadata/v2.py

```diff
@@ -349,6 +349,14 @@ def _default_fill_value(dtype: np.dtype[Any]) -> Any:
         return b""
     elif dtype.kind in "UO":
         return ""
+    elif dtype.kind in "Mm":
```

Comment:
This could use some keen eyes, I've never used these dtypes.

Reply:
I'm also unfamiliar with those dtypes. perhaps you could find examples from the tests on the v2 branch?